src/resource.c

```diff
@@ -959,21 +929,59 @@ hsk_resource_to_dns(const hsk_resource_t *rs, const char *name, uint16_t type) {
   hsk_dns_rrs_t *ns = &msg->ns; // authority
   hsk_dns_rrs_t *ar = &msg->ar; // additional
 
+  // Even though the name here is a single label (the TLD)
+  // we use a larger buffer size of 255 (instead of 63)
+  // to allow escaped byte codes like /000
+  char next[HSK_DNS_MAX_NAME];
+  next_name(tld, next);
```

Comment:
255 is greater than 4 * 63 (252) ;-)

The safest bet is to accommodate a theoretical name that is ALL byte codes. Imagine a 63-character name where each character was `\101` or something. This shouldn't happen in hnsd because we also REFUSE to answer queries for strings with slashes in them, but that's why I think 255 is good to use here instead of 68 or something.

In bns, JJ uses `MAX_NAME_SIZE * 4` for name encoding which is 255 * 4 so it can handle entire URLs. In hnsd resource we will only ever process single label TLD with max 63 characters.

1. I think the math is 62 + 4 actually? The dot and null terminator don't count towards the 63 character count.
2. for NSEC proofs, the name length doesn't matter:
    a. `next_name` is used to prove that we have a domain in the zone but not the record TYPE you asked for
    b. both `next_name` and `prev_name` are used to prove that name just completely does not exist at all

Reply:
I see, in both 2a and 2b above, next_name and prev_name DO NOT actually exist, and those strings ARE the proof that the name you asked for either does (a) or does not (b) exist.

What's funny about this is that typically the names in NSEC proofs DO exist, and that's why this technique is called "black lies": https://blog.cloudflare.com/black-lies/

Reply:
as far as the "dirtyness" - these next/prev names should be the only strings in hnsd that are ever dirty. The code originally checked for dirtyness everytime it wrote any name, which broke these proofs so thats why I refactored the dirty check to JUST cover incoming requests. I asked JJ why he only chose a limited set of characters for that test and havent heard back yet. Probably the "is dirty" test can be replaced with somehting more like hsd rules.verifyName()

Reply:
oh I see..

Yeah, I was just following rfc4470 - I believe it stated those prev/next to be non-existent.. Will check black lies ^

Reply:
Oh right! But that RFC expands on https://datatracker.ietf.org/doc/html/rfc4034#section-4.1.1 where the next name **IS** in the zone. SO it's a nice little trick aka a "lie".